### PR TITLE
e2e jx pipeline

### PR DESCRIPTION
**Pull request by almerico**
e2e test for PR in Jenkinsfile added

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -21,7 +21,9 @@
         steps {
           container('nodejs') {
             sh "npm install"
-            sh "npm run build"
+            //sh "npm run build"
+            sh "npm run e2e"
```
Comment (popovicsandras) [inline Jenkinsfile]: For running e2e you need to set some environment vars to work properly in Jenkins, I assume:
```
API_HOST=""
OAUTH_HOST=""
E2E_HOST="http://localhost"
E2E_PORT="4100"
E2E_USERNAME=""
E2E_PASSWORD=""
E2E_UNAUTHORIZED_USER=""
E2E_UNAUTHORIZED_USER_PASSWORD=""
E2E_FAIL_FAST="true"
BROWSER_RUN="false"
SAVE_SCREENSHOT="true"
SCREENSHOT_URL=""
SCREENSHOT_USERNAME=""
SCREENSHOT_PASSWORD=""
```

and obviously you need a BE for the FE to work with. Do you have it?
@salaboy ?

Comment (almerico) [reply Jenkinsfile]: Hey @popovicsandras, it is in process and I’m added env already but it is still doesn’t work on nodejs builder.

Comment (almerico) [reply Jenkinsfile]: @popovicsandras from https://github.com/Activiti/activiti-modeling-app/blob/master/.travis.yml is not clear how backend is running can you please advice?

Comment (popovicsandras) [reply Jenkinsfile]: @almerico, About how backend is running, I'm afraid I can't tell you too much. We are using our staging environments to test against in Enterprise, either aps2dev or aps2staging.
I would expect to have those staging environments for Community also, like acm2dev or acm2staging?

**Comment by salaboy** [comment]
@almerico can you please share the error of the Chrome web driver.. I will share some of the credential via private channel.

**Comment by almerico** [comment]
`bash-4.2# npm run e2e

> activiti-modeling-app@0.2.0 pree2e /home/jenkins/workspace/iti_activiti-modeling-app_PR-145
> npm run compile-less


> activiti-modeling-app@0.2.0 compile-less /home/jenkins/workspace/iti_activiti-modeling-app_PR-145
> lessc ./node_modules/bpmn-js-properties-panel/styles/properties.less .tmp/bpmn-properties.css


> activiti-modeling-app@0.2.0 e2e /home/jenkins/workspace/iti_activiti-modeling-app_PR-145
> ng e2e

** Angular Live Development Server is listening on localhost:4100, open your browser on http://localhost:4100/ **

Date: 2019-05-15T13:01:03.081Z
Hash: 05a2ead37fa488a0f19e
Time: 43979ms
chunk {dashboard-dashboard-module} dashboard-dashboard-module.js, dashboard-dashboard-module.js.map (dashboard-dashboard-module) 124 kB  [rendered]
chunk {main} main.js, main.js.map (main) 1.16 MB [initial] [rendered]
chunk {polyfills} polyfills.js, polyfills.js.map (polyfills) 774 kB [initial] [rendered]
chunk {runtime} runtime.js, runtime.js.map (runtime) 8.79 kB [entry] [rendered]
chunk {styles} styles.js, styles.js.map (styles) 1.98 MB [initial] [rendered]
chunk {vendor} vendor.js, vendor.js.map (vendor) 14.9 MB [initial] [rendered]
[13:01:03] I/config_source - curl -o/home/jenkins/workspace/iti_activiti-modeling-app_PR-145/node_modules/protractor/node_modules/webdriver-manager/selenium/chrome-response.xml https://chromedriver.storage.googleapis.com/
ℹ ｢wdm｣: Compiled successfully.
[13:01:03] I/update - chromedriver: file exists /home/jenkins/workspace/iti_activiti-modeling-app_PR-145/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_74.0.3729.6.zip
[13:01:03] I/update - chromedriver: unzipping chromedriver_74.0.3729.6.zip
[13:01:04] I/update - chromedriver: setting permissions to 0755 for /home/jenkins/workspace/iti_activiti-modeling-app_PR-145/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_74.0.3729.6
[13:01:04] I/update - chromedriver: chromedriver_74.0.3729.6 up to date
[13:01:05] I/launcher - Running 1 instances of WebDriver
[13:01:05] I/direct - Using ChromeDriver directly...
[13:01:05] E/runner - Unable to start a WebDriver session.
[13:01:05] E/launcher - Error: WebDriverError: unknown error: Chrome failed to start: exited abnormally
  (unknown error: DevToolsActivePort file doesn't exist)
  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
  (Driver info: chromedriver=74.0.3729.6 (255758eccf3d244491b8a1317aa76e1ce10d57e9-refs/branch-heads/3729@{#29}),platform=Linux 4.4.111+ x86_64)
    at Object.checkLegacyResponse (/home/jenkins/workspace/iti_activiti-modeling-app_PR-145/node_modules/selenium-webdriver/lib/error.js:546:15)
    at parseHttpResponse (/home/jenkins/workspace/iti_activiti-modeling-app_PR-145/node_modules/selenium-webdriver/lib/http.js:509:13)
    at doSend.then.response (/home/jenkins/workspace/iti_activiti-modeling-app_PR-145/node_modules/selenium-webdriver/lib/http.js:441:30)
    at processTicksAndRejections (internal/process/next_tick.js:81:5)
Status code is 100, no need to save screenshots.
[13:01:05] E/launcher - Process exited with error code 100
An unexpected error occurred: undefined
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! activiti-modeling-app@0.2.0 e2e: `ng e2e`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the activiti-modeling-app@0.2.0 e2e script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jenkins/.npm/_logs/2019-05-15T13_01_06_023Z-debug.log
bash-4.2#`

**Comment by popovicsandras** [comment]
> Error: WebDriverError: unknown error: Chrome failed to start: exited abnormally
> (unknown error: DevToolsActivePort file doesn't exist)
> (The process stjtarted from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)

Is Chrome installed? You need to have Chrome installed for chrome driver to work.

I don't know how Jenkins X works, but with Travis is kinda easy, we just needed to list Chrome in the addons list of travis.yml:
```
addons:
    chrome: stable
```

Depending on the chosen OS:
https://stackoverflow.com/questions/53073411/selenium-webdriverexceptionchrome-failed-to-start-crashed-as-google-chrome-is

**Comment by almerico** [comment]
our jenkinsx node builder image is here  https://github.com/jenkins-x/jenkins-x-builders/blob/master/builder-nodejs/Dockerfile they are installed  chromedriver chromium xorg-x11-server-Xvfb looks like all that is needed for e2e maybe I missing something

File: e2e/tests/connector/export-connector.e2e.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ import { ConnectorContentPage } from 'ama-testing/e2e';
 
 const path = require('path');
 
-describe('Export connector', () => {
+xdescribe('Export connector', () => {
```
Comment (popovicsandras) [inline e2e/tests/connector/export-connector.e2e.ts]: We should remove this `x`, this way only this test runs. @rdiacenco ?

File: e2e/protractor.conf.js [hunk 3]
```diff
@@ -43,6 +43,7 @@ exports.config = {
             },
             args: [
                 '--incognito',
+                '--no-sandbox',
```
Comment (popovicsandras) [inline e2e/protractor.conf.js]: Why is this needed?

File: Jenkinsfile [hunk 4]
```diff
@@ -12,19 +12,49 @@
       PREVIEW_VERSION = "0.0.0-SNAPSHOT-$BRANCH_NAME-$BUILD_NUMBER"
       NODE_OPTIONS = "--max_old_space_size=40000"
       OPTIMIZE_MEMORY = "true"
+        
+
+      API_HOST="http://activiti-cloud-gateway.jx-staging.35.228.195.195.nip.io/modeling-service/"
```
Comment (popovicsandras) [inline Jenkinsfile]: Don't we want to make these hosts secret as well? In Travis they are, we were requested to remove the urls from the repo because of security concerns. And this seems to be something which could be configured, without the need of making a commit to the repo.

Comment (almerico) [reply Jenkinsfile]: ok

File: e2e/protractor.conf.js (hunk 3 above)
Comment (almerico) [reply e2e/protractor.conf.js]: I added this flag to be able run it on nodejs pod without it I getting 
`[12:46:41] E/runner - Unable to start a WebDriver session.
[12:46:41] E/launcher - Error: WebDriverError: unknown error: Chrome failed to start: exited abnormally
  (unknown error: DevToolsActivePort file doesn't exist)
  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
  (Driver info: chromedriver=74.0.3729.6 (255758eccf3d244491b8a1317aa76e1ce10d57e9-refs/branch-heads/3729@{#29}),platform=Linux 4.4.111+ x86_64)
    at Object.checkLegacyResponse (/home/jenkins/workspace/iti_activiti-modeling-app_PR-145/node_modules/selenium-webdriver/lib/error.js:546:15)
    at parseHttpResponse (/home/jenkins/workspace/iti_activiti-modeling-app_PR-145/node_modules/selenium-webdriver/lib/http.js:509:13)
    at doSend.then.response (/home/jenkins/workspace/iti_activiti-modeling-app_PR-145/node_modules/selenium-webdriver/lib/http.js:441:30)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Status code is 100, no need to save screenshots.
[12:46:41] E/launcher - Process exited with error code 100`

Comment (almerico) [reply e2e/protractor.conf.js]: possibly this can cause issues on travis

File: e2e/tests/connector/export-connector.e2e.ts (hunk 2 above)
Comment (almerico) [reply e2e/tests/connector/export-connector.e2e.ts]: If I uncomment it I got 
`[31m✗ 1. [C286439] Export connector[39m
      [31m- [39m[31mFailed: File /home/jenkins/workspace/iti_activiti-modeling-app_PR-145/e2e-output/downloads/qaconnector.json not found in 10000 msec.
`

Comment (rdiacenco) [reply e2e/tests/connector/export-connector.e2e.ts]: This e2e test fails because of https://github.com/Activiti/Activiti/issues/2773. 
Once we have a fix, we can re-enable export connector e2e test.
